src/backend/joanie/core/admin.py

```diff
@@ -51,11 +68,49 @@ class UserAdmin(auth_admin.UserAdmin):
     list_display = ("username",)
 
 
+class ProductCourseRelationInline(SortableInlineAdminMixin, admin.TabularInline):
+    """Admin class for the ProductCourseRelation model"""
+
+    model = models.Product.target_courses.through
+    extra = 0
+
+
 @admin.register(models.Product)
 class ProductAdmin(TranslatableAdmin):
     """Admin class for the Product model"""
 
     list_display = ("title", "type", "price")
+    fields = (
+        "type",
+        "title",
+        "description",
+        "call_to_action",
+        "price",
+        "certificate_definition",
+        "related_courses",
+    )
+
+    inlines = (ProductCourseRelationInline,)
+    readonly_fields = ("related_courses",)
+
+    @admin.display(description="Related courses")
+    def related_courses(self, obj):  # pylint: disable=no-self-use
+        """
+        Retrieve courses related to the product
+        """
+        if obj.courses.count() > 0:
+            items = [
+                (
+                    "<li>"
+                    f"<a href='{reverse('admin:core_course_change', args=(course.id,),)}'>"
+                    f"{course.code} | {course.title}"
+                    "</a>"
+                    "</li>"
+                )
+                for course in obj.courses.all()
+            ]
```

Comment:
Since you need the data, I would make only one query:
```suggestion
        related_courses = obj.courses.values("id", "code", "title")
        if related_courses:
            items = [
                (
                    "<li>"
                    f"<a href='{reverse('admin:core_course_change', args=(course['id'],),)}'>"
                    f"{course['code']} | {course['title']}"
                    "</a>"
                    "</li>"
                )
                for course in related_courses
            ]
```